guardian/archivehunter
# Feature/Basic version support & S3 SDKv2

## What does this change?

- Adds fields for version support to the main data models
- Updates S3 related code to run on Amazon's SDK v2 instead of v1
- Captures versions on ingest
- Captures versions on bucket scan
- Does NOT apply versioning to proxies

Frontend-facing changes to show the versions will be forthcoming in a seperate PR.

## How to test

Overall functionality should remain unchanged.  
However, on ingestion of a new file you should go to the item view in ArchiveHunter and inspect the data coming from the server in the Network panel of your browser.  You should see the version ID string in the returned data.


## How can we measure success?

Able to track version IDs in the database to improve compatibility with object lock

## Have we considered potential risks?

There are a large number of elements touched by the changes, both functionality wise and in terms of the SDK. We should carefully test this before deployment.
